app/Http/Controllers/Admin/UserController.php

```diff
@@ -69,4 +72,22 @@ public function destroy($id)
 
         return to_route('user.index');
     }
+
+    public function show(string $id)
+    {
+        $user = User::findOrFail($id);
+        // Khởi tạo CourseRepository và sử dụng phương thức getCourses để lấy danh sách khoá học
+        $courseRepository = new CourseRepository(new Course);
```

Comment:
sửa lại cho anh nhé:

use App\Repositories\UserRepository;
use App\Repositories\CourseRepository;
protected $userRepository;
protected $courseRepository;
public function __construct(UserRepository $userRepository, CourseRepository $courseRepository)
{
    $this->userRepository = $userRepository;
    $this->courseRepository = $courseRepository;
}
public function show($id)
{
       return view('users.show', [
            'user' => $this->userRepository->findById($id), // trong BaseRepository em sẽ thêm 1 method findById để lấy data theo id truyền vào 
            'courses' => $this->courseRepository->getCoursesByUserId($id)
        ]);
 ]);